Update `rubocop` to version 1.69.1

mjankowski opened the pull request:
Renovate opened https://github.com/mastodon/mastodon/pull/32508 which needs some fixes first.

This includes a collection of disables/autocorrects.

Not narrowly related to this change, but I'm hoping these issues with the ruby-lsp (interaction between psych and rubocop versions, apparently) - https://github.com/Shopify/ruby-lsp/issues/2840 - might be resolved by this upgrade as well.

Will leave some inline remarks on the autofixes.

mjankowski commented on app/controllers/concerns/cache_concern.rb:
```diff
@@ -28,7 +28,7 @@ def enforce_cache_control!
     raise ArgumentError, 'Only JSON render calls are supported' unless options.key?(:json) || block_given?
 
-    key        = options.delete(:key) || [[params[:controller], params[:action]].join('/'), options[:json].respond_to?(:cache_key) ? options[:json].cache_key : nil, options[:fields].nil? ? nil : options[:fields].join(',')].compact.join(':')
+    key        = options.delete(:key) || [[params[:controller], params[:action]].join('/'), options[:json].respond_to?(:cache_key) ? options[:json].cache_key : nil, options[:fields]&.join(',')].compact.join(':')
```
I think this is safe by itself, but also the `compact` on the end.

The entire second half of this method (how to handle when `options[:key]` isnt present) - should be a separate method.

mjankowski commented on app/lib/request.rb:
```diff
@@ -261,7 +261,6 @@ def open(host, *args)
         outer_e = nil
         port    = args.first
 
-        addresses = []
```
The autocorrect changed this to just be `[]` alone on the line, which is odd?

I do think the assignment is useless though ... will be  assigned via one of the below paths.

mjankowski commented:
Converting to draft - going to attempt to double-check the auto-corrects and/or pre-factor them a bit first.

mjankowski committed:
Fix `Lint/UselessAssignment` cop

mjankowski committed:
Fix `Lint/UselessAssignment` cop

mjankowski commented:
Some linked PRs are going to fail CI because the changes fail w/out also having the rubocop bump. I think my current pref would be to review/merge both of:

- https://github.com/mastodon/mastodon/pull/32970
- https://github.com/mastodon/mastodon/pull/32974

...with those in, I can pull the others into this PR and switch back to Ready state.

mjankowski committed:
Update rubocop to version 1.69.0

mjankowski committed:
Ignore some cops

mjankowski committed:
Fix `Style/MultipleComparison` cop

mjankowski commented:
Rebased after related lint fixes merged, I think ready for review now ... ignoring what I think is a false positive for now.

mjankowski committed:
Ignore `Style/ArrayIntersect` cop

mjankowski committed:
Fix `Style/FillNull` cop

mjankowski committed:
Fix `Style/RedundantRegexpEscape` cop

mjankowski commented:
Ah, 1.69.0 is out now as well -- updated to use that, and updated PR title here.

nschonni commented on .rubocop/style.yml:
```diff
@@ -1,4 +1,7 @@
 ---
+Style/ArrayIntersect:
```
This doesn't seem like a bad idea https://docs.rubocop.org/rubocop/cops_style.html#stylearrayintersect, but maybe there are a lot of hits.
Maybe running `./bin/rubocop --regen-todo` for these, since there likely needs to be a change to that file for the new Rubocop version in the header

mjankowski replied:
Regen'd todo.

mjankowski committed:
Update rubocop to version 1.69.1

mjankowski commented:
Rebased, updated to 1.69.1, re-ran todo generation. No new issues from that patch release bump, still contains things from 1.68.0 and 1.69.0 releases.

mjankowski committed:
wow thats more fun

oneiros commented on app/lib/activitypub/tag_manager.rb:
```diff
@@ -13,7 +13,9 @@ class ActivityPub::TagManager
 
   def public_collection?(uri)
-    uri == COLLECTIONS[:public] || uri == 'as:Public' || uri == 'Public'
+    %w(as:Public Public)
+      .push(COLLECTIONS[:public])
+      .include?(uri)
```
Has this been auto-corrected by rubocop? If yes, I find this baffling.

This is a *lot* less readable than the original and needs to allocate an array where previously none was needed. If this is a new cop, I would say let us please disable it.

mjankowski replied:
I had left this as a note before but may have rebased myself out of it.

The rubocop autocorrect produced:

```
  def public_collection?(uri)
    ['as:Public', 'Public'].include?(uri)
  end
```

Note how that drops the `COLLECTIONS[:public]` from the original.

This is the cop in question - https://docs.rubocop.org/rubocop/cops_style.html#stylemultiplecomparison

The version here was my attempt to fix the autocorrect while still satisfying the cop.

That said - I think I like disable better. Pushed to change to that.

mjankowski replied:
I sure could. https://github.com/rubocop/rubocop/issues/13553

mjankowski committed:
Disable multiple comparison

mjankowski commented:
Rebased after some merge conflicts. Would like to get this merged mostly to resolve the ruby-lsp/psych/rubocop vscode issues, so LMK if any remaining lint-related things to sort out first.
